Add input status to GraphQL API

gligneul opened the pull request:
Close #73

gligneul committed:
feat!: add input status to GraphQL API

Add a new field to the input object in the GraphQL API and a new column
in the Postgres input table.
Node runners should clean their database before upgrading and update the
GraphQL schema.

gligneul committed:
chore: use resolver 2 in Rust

gligneul commented:
Tested it with examples. The input starts with unprocessed, and then it becomes accepted. I rejected the input in the cartesi machine, which also worked as expected.

### Query
```
{
  input(index: 0) {
    index
    status
  }
}
```

### Result
```
{
  "data": {
    "input": {
      "index": 0,
      "status": "UNPROCESSED"
    }
  }
}
```

gligneul commented on offchain/rollups-events/src/rollups_outputs.rs:
```diff
@@ -12,12 +12,31 @@ decl_broker_stream!(RollupsOutputsStream, RollupsOutput, "rollups-outputs");
 }
 
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub struct RollupsAdvanceCompletion {
```
The names are not the same because they have different semantics.

- CompletionStatus is the status of the advance step once it is completed. It doesn't have the unprocessed state.
- AdvanceCompletion is the aggregate of the completion status and the input that generated that status. I couldn't think of a better name for this one.
- InputStatus can be either unprocessed or one of the completion statuses.

gligneul replied:
I think I will rename everything just to CompletionStatus; it might be easier to understand.

marcelstanley replied:
Done at b60f6d6b

torives reviewed (commented):
I found it confusing to create a new type of output to update the status of an advance-state input. 

Creating a new stream for this update event makes more sense to me. `advance-runner` would send the event to it (what is currently `RollupsAdvanceCompletion`) and the `indexer` would listen to it and update the database accordingly.

It's more work, but easier to understand IMO. What do you think?

gligneul commented:
> It's more work, but easier to understand IMO. What do you think?

I agree that it is clearer but it will be much more work. I prefer to keep the PR smaller because all of this will be thrown away when we move to the Go node.

gligneul commented:
@marcelstanley @torives I renamed the enum to CompletionStatus everywhere. I also renamed the AdvanceCompletion to AdvanceResult to clarify that it is an output of the computation instead of an input attribute. The Completion status living inside the input is an implementation detail of the GraphQL/SQL.

gligneul commented on offchain/schema.graphql:
```diff
@@ -1,205 +0,0 @@
-"Data that can be used as proof to validate notices and execute vouchers on the base layer blockchain"
```
This file was committed by accident previously. Now, we are generating it on demand in the CI.
